modules/apps/message-boards/message-boards-web/src/main/resources/META-INF/resources/message_boards/js/MBPortlet.es.js

```diff
@@ -12,225 +12,243 @@
-					.delegate(
-						'click',
-						this.removeAttachment_.bind(this),
-						'.delete-attachment'
-					)
+	Liferay.componentReady(searchContainerId).then((searchContainer) => {
+		searchContainer
+			.get('contentBox')
+			.delegate(
+				'click',
+				removeAttachment.bind(this),
+				'.delete-attachment'
 			);
 
-			this.searchContainer_ = searchContainer;
-		});
+		_searchContainer = searchContainer;
+	});
 
-		const viewRemovedAttachmentsLink = document.getElementById(
-			'view-removed-attachments-link'
-		);
+	const viewRemovedAttachmentsLink = document.getElementById(
+		'view-removed-attachments-link'
+	);
 
-		if (viewRemovedAttachmentsLink) {
-			viewRemovedAttachmentsLink.addEventListener('click', () => {
-				Liferay.Util.openWindow({
-					dialog: {
-						on: {
-							visibleChange: (event) => {
-								if (!event.newVal) {
-									this.updateRemovedAttachments_();
-								}
-							},
-						},
-					},
-					id: this.namespace + 'openRemovedPageAttachments',
-					title: Liferay.Language.get('removed-attachments'),
-					uri: this.viewTrashAttachmentsURL,
-				});
+	if (viewRemovedAttachmentsLink) {
+		viewRemovedAttachmentsLink.addEventListener('click', () => {
+			Liferay.Util.openModal({
+				id: namespace + 'openRemovedPageAttachments',
+				onClose: updateRemovedAttachments,
+				title: Liferay.Language.get('removed-attachments'),
+				url: viewTrashAttachmentsURL,
 			});
-		}
-	}
-
-	/**
-	 * @inheritDoc
-	 */
-
-	detached() {
-		super.detached();
-		this.eventHandler_.removeAllListeners();
+		});
 	}
 
 	/**
 	 * Redirects to the advanced reply page
 	 * keeping the current message.
 	 *
-	 * @protected
 	 */
-
-	openAdvancedReply_() {
-		const inputNode = this.one('#body');
-		inputNode.value = window[
-			this.ns('replyMessageBody' + this.replyToMessageId)
+	const openAdvancedReply = () => {
+		const bodyInput = rootNode.querySelector(`#${namespace}body`);
+		bodyInput.value = window[
+			`${namespace}replyMessageBody${replyToMessageId}`
 		].getHTML();
 
-		const form = this.one(
-			`[name="${this.ns('advancedReplyFm' + this.replyToMessageId)}"]`
+		const form = rootNode.querySelector(
+			`[name="${namespace}advancedReplyFm${replyToMessageId}"]`
 		);
 
 		const advancedReplyInputNode = form.querySelector(
-			`[name="${this.ns('body')}"]`
+			`[name="${namespace}body"]`
 		);
 
-		advancedReplyInputNode.value = inputNode.value;
+		advancedReplyInputNode.value = bodyInput.value;
 
 		submitForm(form);
-	}
+	};
 
 	/**
-	 * Publish the message.
+	 * Sends a request to remove the selected attachment.
 	 *
-	 * @protected
+	 * @param {Event} event The click event that triggered the remove action
 	 */
+	const removeAttachment = (event) => {
+		const link = event.currentTarget;
 
-	publish_() {
-		this.one('#workflowAction').value = this.constants.ACTION_PUBLISH;
-		this.save_();
-	}
+		const deleteURL = link.getAttribute('data-url');
+
+		fetch(deleteURL).then(() => {
+			_searchContainer.deleteRow(
```

Comment:
This is unsafe because there's no guarantee that https://github.com/liferay-frontend/liferay-portal/pull/444/files#diff-0cccf0d3b5fa6eafb0542f34505fdcaf351da1f9de7b67af73dc19e0a26dfcb7R80 has finished and set a value for `_searchContainer`.

Since one is triggered automatically and this is triggered by a user action, the code is very likely to always work well, but the race condition is here nonetheless.